### ISSUE TITLE
Clearing filter clears the selection for a feature

### ISSUE THREAD

**Issue by NyakudyaA** [qgis/QGIS#32147]
**Describe the bug**
I have a polygon layer that I have filtered to show one feature.
![Screenshot 2019-10-07 at 12 33 52](https://user-images.githubusercontent.com/2510900/66305021-c1118980-e8fe-11e9-8c2d-11caba8e387c.png)

I then proceed to select the feature using the visual selection tool
![Screenshot 2019-10-07 at 12 34 44](https://user-images.githubusercontent.com/2510900/66305076-ddadc180-e8fe-11e9-96cc-41feb64f70ee.png)

I then return to clear the filter which in turn clears the selection
![Screenshot 2019-10-07 at 12 35 43](https://user-images.githubusercontent.com/2510900/66305158-03d36180-e8ff-11e9-8764-a26e733f6194.png)

After clearing the selection I see that the feature is no longer selected
![Screenshot 2019-10-07 at 12 36 31](https://user-images.githubusercontent.com/2510900/66305205-1ea5d600-e8ff-11e9-944e-301962c8226d.png)

**QGIS and OS versions**

![Screenshot 2019-10-07 at 12 37 28](https://user-images.githubusercontent.com/2510900/66305281-439a4900-e8ff-11e9-9693-89bf17600f7d.png)

**Comment by elpaso** [comment]
What are you suggesting? A special logic for filter cleared instead of the generic filter-changed behavior?

**Comment by NyakudyaA** [comment]
> What are you suggesting? A special logic for filter cleared instead of the generic filter-changed behavior?

Yes, or maybe just a message to notify a user that clearing the filter will also clear the selection.

**Comment by Pedro-Murteira** [comment]
I can still replicate this on QGIS 3.16.14 and 3.22.1, although it should be considered a feature request instead of a bug.
